sqlglot/expressions.py

```diff
@@ -4738,7 +4738,12 @@ class Extract(Func):
 
 
 class Timestamp(Func):
-    arg_types = {"this": False, "expression": False}
+    arg_types = {"this": False, "expression": False, "with_tz": False}
+
+    @property
+    def return_type(self) -> DataType:
```

Comment:
shouldn't you just override type here?

Reply:
hmm not sure, that kinda leaves room for future bugs if we ever decide for some reason to set the type of Timestamp to something else

Reply:
maybe we can do though `return self._type or ...`

Reply:
should we just have this logic in annotate_types, and then call annotator in the section that you use it?

Reply:
so instead of `expr.return_type` do `annotate_types(expr)` first and then `expr.type` in the other places we need this info?

Reply:
yea, that way we only have typing logic in one place, the annotator

Reply:
that could work, only issue being that we'd annotate the whole subtree of the function call even though we only care about the outermost node

Reply:
ok let's do it that way, it seems like that path shouldn't be hit too frequently anyway so 🤷